x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/BooleanBigArrayBlock.java

```diff
@@ -80,21 +100,21 @@ public BooleanBlock expand() {
             incRef();
             return this;
         }
-        // TODO use reference counting to share the vector
-        try (var builder = blockFactory().newBooleanBlockBuilder(firstValueIndexes[getPositionCount()])) {
-            for (int pos = 0; pos < getPositionCount(); pos++) {
-                if (isNull(pos)) {
-                    builder.appendNull();
-                    continue;
-                }
-                int first = getFirstValueIndex(pos);
-                int end = first + getValueCount(pos);
-                for (int i = first; i < end; i++) {
-                    builder.appendBoolean(getBoolean(i));
-                }
-            }
-            return builder.mvOrdering(MvOrdering.DEDUPLICATED_AND_SORTED_ASCENDING).build();
+        vector.incRef();
+        if (nullsMask == null) {
+            return vector.asBlock();
         }
+        BooleanBigArrayBlock expanded = new BooleanBigArrayBlock(
+            vector,
+            vector.getPositionCount(),
+            null,
+            // TODO: we probably need to adjust the breaker before computing the shifted null mask
```

Comment:
This should be addressed before merging.

Theoretically also applies to the `X-ArrayBlock`s, but the null masks of those are likely much smaller.

Reply:
We could probably just pre-adjust the breaker using the existing null mask to estimate the size of the expanded one.

Reply:
I think the nulls mask will either be `null` itself or will be less than or equal to `the length of the vector / 8`. So it's not *too* bad to pre-adjust and correct the adjustment with the real size.

Reply:
No need to be precise on the pre-adjust. I think. Better to over-allocate and then adjust back down.

Reply:
Alright, I'm using the estimation from BlockRamUsageEstimator, which is essentially `length/8`. That should do the trick.